sdk/eventhub/Azure.Messaging.EventHubs/stress/src/Infrastructure/EventTracking.cs

```diff
@@ -0,0 +1,246 @@
+    ///   to distinct and separate partitions. This is necessary so that missing and duplicate events can be detected by the <see cref="Role.Consumer"/>
+    ///   role without needing access to the <see cref="PartitionPublisher"/> instance.
+    /// </summary>
+    ///
+    /// <param name="partitionCount"></param>
+    /// <param name="roleIndex"></param>
+    /// <param name="partitionIds"></param>
+    ///
+    /// <returns>The array of partitions that a given <see cref="PartitionPublisher"/> should send to.</returns>
+    ///
+    public static List<string> GetAssignedPartitions(int partitionCount, int roleIndex, string[] partitionIds, Role[] roles)
+    {
+        var roleList = new List<Role>(roles);
+
+        var numPublishers = roleList.Where(role => (role == Role.Publisher || role == Role.BufferedPublisher || role == Role.PartitionPublisher)).Count();
+        var thisPublisherIndex = (roleList.GetRange(0,roleIndex)).Where(role => (role == Role.Publisher || role == Role.BufferedPublisher || role == Role.PartitionPublisher)).Count();
+
+        var baseNum = partitionCount/numPublishers;
+        var remainder = partitionCount % numPublishers;
+
+        var startPartition = 0;
+        var endPartition = 0;
+
+        if (thisPublisherIndex >= remainder)
+        {
+            startPartition = (baseNum*thisPublisherIndex) + remainder;
+            endPartition = startPartition + baseNum;
+        }
+        else
+        {
+            startPartition = (baseNum*thisPublisherIndex) + thisPublisherIndex;
+            endPartition = startPartition + baseNum + 1;
+        }
+
+        var assignedPartitions = new List<string>();
+
+        for (int i = startPartition; i < endPartition; i++)
+        {
+            assignedPartitions.Add(partitionIds[i]);
+        }
+
+        return assignedPartitions;
+    }
+
+    /// <summary>
+    ///   Validates an event. This instance checks if the event has an id, if the event has already been seen before by this instance, that the
+    ///   intended partition was the one the event was received from, that the event was received in order, and that none were missed, and that
+    ///   the body is valid.
+    /// </summary>
+    ///
+    /// <param name="eventData"></param>
+    /// <param name="partitionReceivedFrom"></param>
+    /// <param name="metrics"></param>
+    ///
+    /// <returns>The publisher assigned index number of the event that was just read, or -1 if this event was unkown or a duplicate.</returns>
+    ///
+    /// <remarks>
+    ///   One instance should be created for each processor or consumer role. This instance will be able to keep track of all the events the
+    ///   processor or consumer has received for each partition it is reading or processing from.
+    /// </remarks>
+    ///
+    private int _checkEvent(EventData eventData, string partitionReceivedFrom, Metrics metrics)
```

Comment:
```suggestion
    private int CheckEvent(EventData eventData, string partitionReceivedFrom, Metrics metrics)
```

nit:  The underscore convention is only used for class-level variables.